src/main/scala/org/deepdive/inference/datastore/PostgresInferenceDataStore.scala

```diff
@@ -104,5 +104,95 @@ trait PostgresInferenceDataStoreComponent extends SQLInferenceDataStoreComponent
       // do nothing
     }
 
+    // assign senquential ids to table's id column
+    def assignIds(table: String, startId: Long, sequence: String) : Long = {
+      if (ds.isUsingGreenplum()) {
+        executeQuery(s"SELECT fast_seqassign('${table.toLowerCase()}', ${startId});");
+      } else {
+        execute(s"UPDATE ${table} SET id = ${nextVal(sequence)};")
+      }
+      var count : Long = 0
+      issueQuery(s"""SELECT COUNT(*) FROM ${table};""") { rs =>
+        count = rs.getLong(1)
+      }
+      return count
+    }
+    
+    // create fast sequence assign function for greenplum
+    def createAssignIdFunctionGreenplum() : Unit = {
+      if (!ds.isUsingGreenplum()) return
+
+      val sql = """
```

Comment:
This kind of construction makes me extremely nervous.  It's basically a huge amount of code that's entirely opaque to the build system; what if there's a stray keystroke here?  We won't know until something fails, weirdly.  

Can we move these function definitions into their own Scala wrapper objects, with corresponding isolated test code?